### DISCUSSION TITLE
Enable Smart Search Features for Shared Album Recipients

### DISCUSSION THREAD

**Discussion by RaulGw4** [immich-app/immich#10579] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi everyone. I wanted to refine my Feature Request with a more thorough explanation, so here it is:

**Description:**

Currently, Immich allows users to take advantage of advanced machine learning features such as **facial recognition**, **object search**, and various **metadata searches** (by date, location, camera type, media type, etc.). However, these features are only available to the person who uploads the photos. I propose adding the ability for shared album recipients to also access these features.

**Proposed Feature:**

1. **🔄 Toggle Switch for Shared Album Settings:**
   - When sharing an album, provide a toggle switch labeled **"Grant Recipient Access to Smart Search"**.

https://github.com/user-attachments/assets/30859dac-db0c-4c56-b9e5-958549a2b09f

   - This switch will grant the shared album recipients access to machine learning features, allowing them to:
     - Use **facial recognition** to find photos with specific people.
     - Search for **objects** within the photos.
     - Utilize **metadata searches** (by date, location, camera type, media type, etc.).

**Conclusion:**
Implementing this feature will significantly enhance the functionality and user experience of Immich, making it a more powerful and user-friendly tool for managing and sharing photos.

> **👏 Cheers to all who have worked on this amazing project! Great job to the Immich team on all the incredible work you have done so far! This feature would be a fantastic addition and would greatly enhance the app's usability for everyone. Keep up the excellent work!**

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Scorcerer** [comment]
I was hitting the same issue but with shared partner libraries:
I have a bunch of old, imported photos, and only one user has them imported into their account (let's call it OldPhotos). Now, this user shares its library with other users. Unfortunately, seeing the faces in my account is impossible, even though the OldPhotos has them just fine.

**Comment by ghost** [comment]
Indeed, sharing people would be a great feature.

**Comment by bo0tzz** [comment]
#2995

**Comment by ghost** [comment]
I've seen this discussion was edited, this would be an even better and generalized solution.
